src/cpp/session/modules/SessionFind.cpp

```diff
@@ -741,6 +719,38 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
       return error;
    }
 
+   void cleanLineAndGetMatches(std::string* pEncodedLine,
+                               json::Array* pMatchOn, json::Array* pMatchOff)
+   {
+      // The incoming string is assumed to have color encodings from the initial grep command.
+      // These encodings are parsed out and their positions are placed in pMatchOn and pMatchOff.
+
+      const char* inputPos = pEncodedLine->c_str();
+      const char* end = inputPos + pEncodedLine->size();
+      std::size_t charactersProcessed = 0;
+      boost::cmatch match;
+      std::string cleanLine;
+
+      while (regex_utils::search(inputPos, match, getColorEncodingRegex(findResults().gitFlag())))
+      {
+         std::string matchedString(inputPos, inputPos + match.position());
+         inputPos += match.position() + match.length();
+         
+         cleanLine.append(matchedString);
+
+         charactersProcessed += matchedString.size();
+   
+         if ((match.size() > 2 && match[2] == "1" && findResults().gitFlag()) ||
```

Comment:
I think these string literals should have a comment explaining what they mean/where they come from.

Reply:
Ok, updated.